qucs-doc/report/EKV26/Makefile.am

```diff
@@ -29,6 +29,8 @@ EXTRA_DIST = $(SOURCES)
 SOURCES = EKV26.tex content.tex
 TARGETS = EKV26.pdf
 
+export TEXINPUTS=$(srcdir):$(srcdir)/..:
```

Comment:
> we can't do that [passing include flags to latex]

yes, we *can do* that. just unfortunately, we have to pass through the environment, because latex does not support `-I`. this is exactly what happens in the current version, except the author didn't know about that colon trick.

> having automake automatically prepared  ..  but that's not possible

anything is *possible*. using an include file certainly is, and that's practically equivalent.

> understanding what the rule is doing, 

yes. if the rule is "run program with arguments", it should mention 1) the program 2) the arguments, for my understanding.

> yes, you said that already

sorry i meant to emphasize the "exporting" in "exporting TEXINPUTS (globally)" [is bad], since this is not so much about "TEXINPUTS"...

the (hopefully better) explanation is: argument passing is *local*, so it should happen locally. a local inline export exactly implements argument passing in this [latex] case (i.e. is functionally equivalent). in contrast, a global export is 1) not visible where it should and 2) may even leak to the wrong commands.

> To me all this, until now, looks like a difference of style only.

maybe. the wiki articles on programming/coding style i found, do not mention variable scopes at all. style seems to be about indentation and whitespace, also 'naming conventions'. i think this discussion is more about semantics.

but there's some relief [here](http://c2.com/cgi/wiki?GlobalVariablesAreBad)

> For experienced programmers, there is little temptation. Using a local or the SingletonPattern is usually easy, even if it does take a few extra seconds to type. After you get burned a couple of times, you will forget the notion that globals would make things easier.